src/components/TradeWidget/TokenRow.tsx

```diff
@@ -110,14 +111,22 @@ const customSelectStyles = {
   valueContainer: (provided: CSSProperties): CSSProperties => ({ ...provided, minWidth: '4.5em' }),
 }
 
+function displayBalance(balance: TokenBalanceDetails | undefined, key: string): string {
+  if (balance === undefined) {
```

Comment:
nit: I think we don't have null checks in typescript yet, so I think we could get an error if balance is null (`null[key]`). I think it's fine and shorter just testing for falsy values:

```js
if (!balance) {
  return '0' 
}
```